src/Support/PresetProvider.php

```
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Peck\Support;
+
+/**
+ * Simple helper to provide the whitelisted words for a given preset.
+ * The whitelisted words are used to ignore certain words when spellchecking.
+ */
+final readonly class PresetProvider
+{
+    private const string PRESET_STUBS_DIRECTORY = '/../../stubs/presets';
+
+    /**
+     * Returns the whitelisted words for the given preset.
+     */
+    public static function whitelistedWords(?string $preset): array
+    {
+        if ($preset === null) {
+            return [];
+        }
+
+        $path = self::getStubPath($preset);
+
+        if (! file_exists($path)) {
+            return [];
```

I think that instead of just returning an empty array, we can inform the user that the chosen preset is invalid.

---

Will take a look tomorrow. Wasn't sure if we want to throw an exception or not etc.